doc/usage/restructuredtext/domains.rst

```diff
@@ -1739,6 +1739,8 @@ There is a set of directives allowing documenting command-line programs:
 
    then ``:option:`rm -r``` would refer to the first option, while
    ``:option:`svn -r``` would refer to the second one.
+   Note using ``:option:`-r <rm -r>``` would refer the first option, while
+   display text would be only ``-r``.
```

Comment:
This notation is not only for the `:option:` role, but for other cross-reference roles.
And it has been already described at the top of this page:
https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#cross-referencing-syntax (the first item of the bullet list)

So -0 for adding this.